### PR TITLE
Update base survey objects for serialized data

### PR DESCRIPTION
**Pull request by kaligrafy**

### CODE REVIEW

File: packages/evolution-common/src/utils/DateUtils.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// can return invalid dates: use isNaN on date.getTime() to check if it's valid
```
Comment (tahini) [inline packages/evolution-common/src/utils/DateUtils.ts:1]: Missing copyright header

File: packages/evolution-common/src/services/baseObjects/BasePerson.ts [hunk 2]
```diff
@@ -274,57 +249,6 @@ class BasePerson extends Uuidable implements IBasePersonAttributes, IValidatable
...
-        if (
-            dirtyParams.baseTrips !== undefined &&
-            (!Array.isArray(dirtyParams.baseTrips) || !dirtyParams.baseTrips.every((trip) => trip instanceof BaseTrip))
-        ) {
-            errors.push(new Error('BasePerson validateParams: baseTrips should be an array of BaseTrip'));
-        }
-
-        // Validate baseVehicles (if provided)
-        if (
-            dirtyParams.baseVehicles !== undefined &&
```
Comment (tahini) [inline packages/evolution-common/src/services/baseObjects/BasePerson.ts:321]: Where are those validations now?

Comment (kaligrafy) [reply packages/evolution-common/src/services/baseObjects/BasePerson.ts:321]: they were never used, this was base object composition, which should not happen. We will composed only custom objects in survey itself.

File: packages/evolution-common/src/utils/DateUtils.ts (hunk 1 above)
Comment (tahini) [reply packages/evolution-common/src/utils/DateUtils.ts:11]: Please add unit tests for this file, it's a util after all!

**Review by tahini — CHANGES_REQUESTED** [review]

**Review by tahini — APPROVED** [review]

**Review by tahini — APPROVED** [review]

**Review by tahini — APPROVED** [review]

**Pull request merged**